deployment/flexible_api_server.py

```
@@ -0,0 +1,564 @@
+                return self._predict_local(text)
+
+        except Exception as e:
+            # Log error with redacted text for debugging (avoid PII exposure in logs)
+            text_preview = f"{text[:20]}..." if len(text) > 20 else text
+            logger.error(f"❌ Prediction failed: {e} (input preview: {text_preview})")
+            return {
+                "error": str(e),
+                "deployment_type": self.deployment_type.value
+            }
+
+    def _predict_serverless(self, text: str) -> Dict[str, Any]:
```

Ruff (0.12.2)

Use `dict` instead of `Dict` for type annotation

Replace with `dict`

(UP006)